browser/ui/views/frame/vertical_tab_strip_region_view.cc

```diff
@@ -0,0 +1,285 @@
+}
+
+void VerticalTabStripRegionView::Layout() {
+  // As we have to update ScrollView's viewport size and its contents size,
+  // layouting children manually will be more handy.
+
+  // 1. New tab should be fixed at the bottom of container.
+  auto contents_bounds = GetContentsBounds();
+  new_tab_button_->SetSize(
+      {contents_bounds.width(), new_tab_button_->GetPreferredSize().height()});
+  new_tab_button_->SetPosition(
+      {contents_bounds.x(),
+       contents_bounds.bottom() - new_tab_button_->height()});
+
+  // 2. ScrollView takes the rest of space.
+  // Set preferred size for scroll view to know this.
+  const gfx::Size header_size{scroll_view_->width(),
+                              new_tab_button_->GetPreferredSize().height()};
+  scroll_view_header_->SetPreferredSize(header_size);
+  scroll_view_header_->SetSize(header_size);
+  scroll_view_->SetSize({contents_bounds.width(),
+                         contents_bounds.height() - new_tab_button_->height()});
+  scroll_view_->SetPosition(contents_bounds.origin());
+  scroll_view_->ClipHeightTo(0, scroll_view_->height() - header_size.height());
+
+  auto* scroll_view_contents = scroll_view_->contents();
+  scroll_view_contents->SetSize(
+      {scroll_view_->width(), scroll_view_contents->height()});
+
+  UpdateNewTabButtonVisibility();
+  UpdateTabSearchButtonVisibility();
+}
+
+void VerticalTabStripRegionView::UpdateLayout() {
+  if (tabs::features::ShouldShowVerticalTabs()) {
+    if (!Contains(region_view_)) {
+      original_parent_of_region_view_ = region_view_->parent();
+      original_parent_of_region_view_->RemoveChildView(region_view_);
+      scroll_view_->contents()->AddChildView(region_view_.get());
+    }
+    region_view_->layout_manager_->SetOrientation(
+        views::LayoutOrientation::kVertical);
+  } else {
+    if (Contains(region_view_)) {
+      scroll_view_->contents()->RemoveChildView(region_view_);
+      original_parent_of_region_view_->AddChildView(region_view_.get());
+    }
+    region_view_->layout_manager_->SetOrientation(
+        views::LayoutOrientation::kHorizontal);
+  }
+  Layout();
+}
+
+void VerticalTabStripRegionView::OnThemeChanged() {
+  View::OnThemeChanged();
+
+  auto background_color =
+      GetColorProvider()->GetColor(ThemeProperties::COLOR_TOOLBAR);
+  SetBackground(views::CreateSolidBackground(background_color));
+  scroll_view_->SetBackgroundColor(background_color);
+
+  new_tab_button_->FrameColorsChanged();
+}
+
+void VerticalTabStripRegionView::UpdateNewTabButtonVisibility() {
+  bool overflowed =
+      scroll_view_->GetMaxHeight() < scroll_view_->contents()->height();
+  region_view_->new_tab_button()->SetVisible(!overflowed);
+  new_tab_button_->SetVisible(overflowed);
+}
+
+void VerticalTabStripRegionView::UpdateTabSearchButtonVisibility() {
+  const bool is_vertical_tabs = tabs::features::ShouldShowVerticalTabs();
+  if (auto* tab_search_button = region_view_->tab_search_button())
```

Comment:
We will need to figure out where this goes as it is a useful feature.

Reply:
Yeah, on Windows, the search button will remain in frame area even with vertical tabs. This only happens on MacOS. I think vertical tabs' header area is a good place to put this button!